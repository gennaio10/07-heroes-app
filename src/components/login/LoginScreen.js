import React, { useContext } from "react";
import { AuthContext } from "../../auth/AuthContext";
import { types } from "../../types/types";

export const LoginScreen = ({ history }) => {
  const { dispatch } = useContext(AuthContext);

  const handleLogin = () => {
    // history.push('/');

    const lastPath = localStorage.getItem('lastPath') || '/';

    const action = {
      type: types.login,
      payload: { name: "Pororo" },
    };

    dispatch(action);
    history.replace(lastPath);
    // history.replace("/");
  };

  return (
    <div className="container">
      <br />
      <br />
      <br />
      <br />
      <br />
      <div className="d-flex justify-content-center h-100">
        <div className="card">
          <div className="card-header">
            <h3>Sign In</h3>
          </div>
          <div className="card-body">
            <div>
              <div className="input-group form-group">
                <div className="input-group-prepend">
                  <span className="input-group-text">
                    <i className="fas fa-user"></i>
                  </span>
                </div>
                <input
                  type="text"
                  className="form-control"
                  placeholder="username"
                />
              </div>
              <div className="input-group form-group">
                <div className="input-group-prepend">
                  <span className="input-group-text">
                    <i className="fas fa-key"></i>
                  </span>
                </div>
                <input
                  type="password"
                  className="form-control"
                  placeholder="password"
                />
              </div>
              <div className="form-group">
                <button className="btn btn-primary" onClick={handleLogin}>
                  Login
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
