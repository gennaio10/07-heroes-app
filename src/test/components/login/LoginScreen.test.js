import React from "react";
import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { AuthContext } from "../../../auth/AuthContext";
import { types } from "../../../types/types";
import { LoginScreen } from "../../../components/login/LoginScreen";

describe("Pruebas de <LoginScreen />)", () => {
  const defaultName = "Pororo";

  const historyMock = {
    replace: jest.fn(),
  };

  const contextValue = {
    dispatch: jest.fn(),
    user: { name: defaultName },
  };

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <LoginScreen history={historyMock} />
    </AuthContext.Provider>
  );

  test("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("button").text().trim()).toBe("Login");
  });

  test("debe de realizar el dispatch y la navegacion", () => {
    const handleLogin = wrapper.find("button").prop("onClick");
    handleLogin();

    expect(contextValue.dispatch).toHaveBeenCalledTimes(1);
    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.login,
      payload: { name: defaultName },
    });
    expect(historyMock.replace).toHaveBeenCalledTimes(1);
    expect(historyMock.replace).toHaveBeenCalledWith("/");

    const lastPathEsperado = '/dc';
    localStorage.setItem('lastPath', lastPathEsperado);
    handleLogin();

    expect(historyMock.replace).toHaveBeenCalledWith(lastPathEsperado);
  });
});
