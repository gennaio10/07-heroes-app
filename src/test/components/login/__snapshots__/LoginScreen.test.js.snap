// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <LoginScreen />) debe de mostrarse correctamente 1`] = `
<div
  className="container"
>
  <br />
  <br />
  <br />
  <br />
  <br />
  <div
    className="d-flex justify-content-center h-100"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        <h3>
          Sign In
        </h3>
      </div>
      <div
        className="card-body"
      >
        <div>
          <div
            className="input-group form-group"
          >
            <div
              className="input-group-prepend"
            >
              <span
                className="input-group-text"
              >
                <i
                  className="fas fa-user"
                />
              </span>
            </div>
            <input
              className="form-control"
              placeholder="username"
              type="text"
            />
          </div>
          <div
            className="input-group form-group"
          >
            <div
              className="input-group-prepend"
            >
              <span
                className="input-group-text"
              >
                <i
                  className="fas fa-key"
                />
              </span>
            </div>
            <input
              className="form-control"
              placeholder="password"
              type="password"
            />
          </div>
          <div
            className="form-group"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
            >
              Login
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
